src/dom-to-image-more.js

```diff
@@ -74,22 +75,46 @@
         const ownerWindow = domtoimage.impl.util.getWindow(node);
         options = options || {};
         copyOptions(options);
+        let restorations = [];
         return Promise.resolve(node)
+            .then(ensureElement)
             .then(function (clonee) {
                 return cloneNode(clonee, options, null, ownerWindow);
             })
             .then(embedFonts)
             .then(inlineImages)
             .then(applyOptions)
-            .then(function (clone) {
-                return makeSvgDataUri(
-                    clone,
-                    options.width || util.width(node),
-                    options.height || util.height(node)
-                );
-            })
+            .then(makeSvgDataUri)
+            .then(restoreWrappers)
             .then(clearCache);
 
+        function ensureElement(node) {
+            if (node.nodeType === ELEMENT_NODE) return node;
+
+            const originalChild = node;
+            const originalParent = node.parentNode;
+            const wrappingSpan = document.createElement('span');
+            originalParent.replaceChild(wrappingSpan, originalChild);
+            wrappingSpan.append(node);
+            restorations.push({
+                parent: originalParent,
+                child: originalChild,
+                wrapper: wrappingSpan,
+            });
+            return wrappingSpan;
+        }
+
+        function restoreWrappers(result) {
+            // put the original children back where the wrappers were inserted
+            while (restorations.length > 0) {
+                const restoration = restorations.pop();
+                //console.log(`parent: ${restoration.parent}\nchild: ${restoration.child}\nwrapper: ${restoration.wrapper}`);
+                restoration.parent.replaceChild(restoration.child, restoration.wrapper);
```

Comment:
Rip out our wrapping `span`. I *wonder* if we need to remove the wrapper or something else? But seems fine. Will need to know if this change breaks anything for _real_ consumers, hope #62 can test and report back.

Reply:
@meche_gh this is the biggest questionable area... is there anything I need to do the swap-back better?